app/src/org/commcare/android/framework/SessionAwareFragmentActivity.java

```diff
@@ -0,0 +1,76 @@
+package org.commcare.android.framework;
+
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.content.IntentFilter;
+import android.os.Bundle;
+import android.support.v4.app.FragmentActivity;
+
+import org.commcare.dalvik.activities.LoginActivity;
+
+/**
+ * Manage redirection to login screen when session expiration occurs.
+ *
+ * @author Phillip Mates (pmates@dimagi.com)
+ */
+public abstract class SessionAwareFragmentActivity extends FragmentActivity {
```

Comment:
@ctsims I created this fragment activity that checks to see if the session has expired and the user needs to redirect to the login screen. 

I've made CommCareActivity extend it. Can you think of any other classes that extend FragmentActivity that would need to extend it?